packaging/os/yum.py

```diff
@@ -646,60 +698,40 @@ def install(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos):
                 if is_installed(module, repoq, spec, conf_file, en_repos=en_repos, dis_repos=dis_repos):
                     found = True
                     res['results'].append('package providing %s is already installed' % (spec))
-                    
+
             if found:
                 continue
 
+
+            # downgrade - the yum install command will only install or upgrade to a spec version, it will
+            # not install an older version of an RPM even if specified by the install spec. So we need to
+            # determine if this is a downgrade, and then use the yum downgrade command to install the RPM.
+
+            # Get the NEVRA of the requested package
+            (name, ver, rel, epoch, arch) = splitFilename(pkglist.pop())
```

Comment:
Why use pkglist.pop() insgtead of pkglist[0]?

is pkglist[0] always going to be the name that was specified?  What if the name that was specified is a virtual provide?  ie: Package named foo at version 2.0 has Provides: bar = 1.5.  If the ansible command specifies name: "bar=1.0" will pkglist[0] be foo-2.0 or bar-1.5?